lib/modulesync/puppet_module.rb

```diff
@@ -1,49 +1,37 @@
-require 'modulesync'
-require 'modulesync/util'
+require 'puppet_blacksmith'
 
-module ModuleSync
-  # Provide methods to retrieve puppet module attributes
-  class PuppetModule
-    attr_reader :given_name
-    attr_reader :options
-
-    def initialize(given_name, options)
-      options ||= {}
-      @options = Util.symbolize_keys(options)
-
-      @given_name = given_name
-
-      return unless given_name.include?('/')
-
-      @repository_name = given_name.split('/').last
-      @repository_namespace = given_name.split('/')[0...-1].join('/')
-    end
-
-    def repository_name
-      @repository_name ||= given_name
-    end
+require 'modulesync/source_code'
 
-    def repository_namespace
-      @repository_namespace ||= @options[:namespace] || ModuleSync.options[:namespace]
-    end
-
-    def repository_path
-      @repository_path ||= "#{repository_namespace}/#{repository_name}"
-    end
-
-    def repository_remote
-      @repository_remote ||= @options[:remote] || _repository_remote
-    end
-
-    def working_directory
-      @working_directory ||= File.join(ModuleSync.options[:project_root], repository_path)
+module ModuleSync
+  # Provide methods to manipule puppet module code
+  class PuppetModule < SourceCode
+    def update_changelog(version, message)
+      changelog = "#{working_directory}/CHANGELOG.md"
```

Comment:
Perhaps use `File.join` while you're at it?
```suggestion
      changelog = File.join(working_directory, 'CHANGELOG.md')
```

Reply:
Like you mentioned previously, I turned `ModuleSync::module_file` into `SourceCode#path`, then use it here.
Thanks.